src/plots/polar/polar.js

```diff
@@ -1116,26 +1278,229 @@ function computeSectorBBox(sector) {
+    var s1 = wrap360(sector[1]);
+    if(s0 > s1) s1 += 360;
+
+    var deg = wrap360(rad2deg(rad));
+    var nextTurnDeg = deg + 360;
+
+    return (deg >= s0 && deg <= s1) ||
+        (nextTurnDeg >= s0 && nextTurnDeg <= s1);
+}
+
+function snapToVertexAngle(a, vangles) {
+    function angleDeltaAbs(va) {
+        return Math.abs(angleDelta(a, va));
     }
 
-    var xs = r * Math.cos(deg2rad(sector[0]));
-    var ys = -r * Math.sin(deg2rad(sector[0]));
-    var xe = r * Math.cos(deg2rad(sector[1]));
-    var ye = -r * Math.sin(deg2rad(sector[1]));
+    var ind = findIndexOfMin(vangles, angleDeltaAbs);
+    return vangles[ind];
+}
 
-    var arc = Math.abs(sector[1] - sector[0]);
-    var flags = arc <= 180 ? [0, 0, 0] : [0, 1, 0];
+// taken from https://stackoverflow.com/a/2007279
+function angleDelta(a, b) {
+    var d = b - a;
+    return Math.atan2(Math.sin(d), Math.cos(d));
+}
+
+function findIndexOfMin(arr, fn) {
+    fn = fn || Lib.identity;
+
+    var min = Infinity;
+    var ind;
+
+    for(var i = 0; i < arr.length; i++) {
+        var v = fn(arr[i]);
+        if(v < min) {
+            min = v;
+            ind = i;
+        }
+    }
+    return ind;
+}
+
+// find intersection of 'v0' <-> 'v1' edge with a 'radial' line
+// (i.e. a line that starts from the origin at angle 'a')
+// given an (xp,yp) pair on the 'v0' <-> 'v1' line
+// (N.B. 'v0' and 'v1' are angles in radians)
+function findIntersectionXY(v0, v1, a, xpyp) {
+    var xstar, ystar;
+
+    var xp = xpyp[0];
+    var yp = xpyp[1];
+    var dsin = clampTiny(Math.sin(v1) - Math.sin(v0));
+    var dcos = clampTiny(Math.cos(v1) - Math.cos(v0));
+
+    if(dsin && dcos) {
+        // given
+        //  g(x) := v0 -> v1 line = m * x + b
+        //  h(x) := mr * x
+        // solve g(xstar) = h(xstar)
+        var m = dsin / dcos;
+        var b = yp - m * xp;
+        var mr = Math.sin(a) / Math.cos(a);
+        xstar = b / (mr - m);
+        ystar = mr * xstar;
+    } else {
+        var r;
+        if(dcos) {
+            // horizontal v0 -> v1
+            r = yp / Math.sin(v0);
+        } else {
+            // vertical v0 -> va
+            r = xp / Math.cos(v0);
+        }
+        xstar = r * Math.cos(a);
+        ystar = r * Math.sin(a);
+    }
+
+    return [xstar, ystar];
+}
+
+// solves l^2 = (f(x)^2 - yp)^2 + (x - xp)^2
+// rearranged into 0 = a*x^2 + b * x + c
+//
+// where f(x) = m*x + t + yp
+// and   x01 = (-b +/- del) / (2*a)
+function findXYatLength(l, m, xp, yp) {
+    var t = -m * xp;
+    var a = m * m + 1;
+    var b = 2 * (m * t - xp);
+    var c = t * t + xp * xp - l * l;
+    var del = Math.sqrt(b * b - 4 * a * c);
+    var x0 = (-b + del) / (2 * a);
+    var x1 = (-b - del) / (2 * a);
+    return [
+        [x0, m * x0 + t + yp],
+        [x1, m * x1 + t + yp]
+    ];
+}
+
+function makeCycleIndexFn(len) {
+    return function(index) {
+        return index < 0 ? len + index :
+            index < len ? index : index - len;
+    };
```

Comment:
[`Lib.mod`](https://github.com/plotly/plotly.js/blob/master/src/lib/mod.js)?

Reply:
Oh, so nice. Thanks for tip. Sub'ed in -> https://github.com/plotly/plotly.js/pull/2739/commits/1ca65080d138a42442197b9389c04ab9b0cdbeca